modules/xfem/src/dirackernels/XFEMIntegratedBC.C

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "XFEMIntegratedBC.h"
+#include "ElementPairLocator.h"
+
+InputParameters
+XFEMIntegratedBC::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+  params.addClassDescription("Applies a pressure on an interface cut by XFEM.");
+  params.addRequiredParam<UserObjectName>("geometric_cut_userobject",
+                                          "Name of GeometricCutUserObject associated with this BC");
+  return params;
+}
+
+XFEMIntegratedBC::XFEMIntegratedBC(const InputParameters & parameters)
+  : DiracKernel(parameters),
+    _gcuo(getUserObject<GeometricCutUserObject>("geometric_cut_userobject")),
+    _interface_id(_gcuo.getInterfaceID()),
+    _element_pair_locators(_subproblem.geomSearchData()._element_pair_locators)
+{
+}
+
+void
+XFEMIntegratedBC::addPoints()
+{
+  _elem_qp_normal.clear();
+  _elem_qp_JxW.clear();
+
+  // Get element pairs on the interface
+  const ElementPairLocator & elem_pair_loc = *_element_pair_locators.at(_interface_id);
+
+  // Go over element pairs
+  auto elem_pairs = elem_pair_loc.getElemPairs();
+  for (auto elem_pair : elem_pairs)
+  {
+    const Elem * elem1 = elem_pair.first;
+    const Elem * elem2 = elem_pair.second;
+    const ElementPairInfo & info = elem_pair_loc.getElemPairInfo(elem_pair);
+
+    for (unsigned int i = 0; i < info._elem1_constraint_q_point.size(); ++i)
+    {
+      _elem_qp_normal[elem1][i] = info._elem1_normal;
```

Comment:
@jiangwen84 In your original code, you had `_elem_qp_normal[elem1][i] = -info._elem1_normal;` with a minus sign. I removed the minus sign so that it becomes the exterior (outward pointing) normal on the embedded element edge. I think this convention is consistent with all other objects in tensor_mechanics.

As a result I also flipped the sign in XFEMPressure.C. Could you help me double check?

Reply:
I was not sure why I had a minus sign in my code. Now I think it makes more sense to remove it. 

Your changes look good to me.